unittests/allplatformstests.py

```diff
@@ -4263,3 +4263,16 @@ def test_custom_target_name(self):
         if self.backend is Backend.ninja:
             self.assertIn('Generating file.txt with a custom command', out)
             self.assertIn('Generating subdir/file.txt with a custom command', out)
+
+    def test_symlinked_subproject(self):
+        testdir = os.path.join(self.unit_test_dir, '106 subproject symlink')
+        subproject = os.path.join(testdir, 'symlinked_subproject')
+        symlinked_subproject = os.path.join(testdir, 'subprojects', 'symlinked_subproject')
+        if os.path.exists(symlinked_subproject):
+            os.remove(symlinked_subproject)    
+        os.symlink(subproject, symlinked_subproject)
+
```

Comment:
The better thing to do here is:
```python
os.symlink(subproject, symlinked_subproject)
self.addCleanup(os.remove, symlinked_subproject)
```

Which is guaranteed to get called even if there's an exception in `self.init` or `self.build`, unlike the current code which needs the check to ensure that the symlink was removed in the last run.